driver/src/price_estimation/average_price_source.rs

```diff
@@ -0,0 +1,86 @@
+use super::{PriceSource, Token};
+use crate::models::TokenId;
+use anyhow::{anyhow, Result};
+use crossbeam_utils::thread;
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+/// Combines two prices into one average.
+pub struct AveragePriceSource<T0, T1> {
```

Comment:
If T0 and T1 are both PriceSource + Send, can we just make it a statically sized array so we can iterate over it? Otherwise, as we grow I'm afraid the match logic below will explode in combinations. 
But we can also cross that bridge when we get there.

Reply:
ubernit: maybe AveragingPriceSource is a slightly better name?

Reply:
I agree that if we are going to add more price sources in the future and average all of them then we have to make this a list.
Until then this is simpler so let's not do this before this is necessary.